packages/venia-concept/src/ServiceWorker/Utilities/htmlHandler.js

```diff
@@ -6,12 +10,43 @@ const cloneRequestWithDiffURL = (request, url) =>
           })
         : request;
 
+/**
+ * cacheHTMLPlugin is a workbox plugin that will apply request
+ * and reponse manipulations on HTML routes.
+ */
 export const cacheHTMLPlugin = {
     cacheKeyWillBeUsed: async () => {
+        /**
+         * For all read and writes that arrive on any HTML route
+         * return `/`.
+         */
         return '/';
     },
     requestWillFetch: async ({ request }) => {
+        /**
+         * Clone the request with url as `/`.
+         */
         const newRequest = cloneRequestWithDiffURL(request, '/');
+
         return newRequest;
+    },
+    fetchDidSucceed: async ({ request, response }) => {
+        /**
+         * Check if there is a response cached for the request
+         * URL. If yes, compare the content of both the cached
+         * response and the new response. If they are different
+         * send a the `HTML_UPDATE_AVAILABLE` message to window.
+         */
+        const cachedResponseObj = await caches.match(
```

Comment:
Does `fetchDidSucceed` get called before or after the response is put into `caches`?

Reply:
Before the cache replacement is done. Good question.